### PR TITLE
rados: add support for listing a specific pg

### PR DESCRIPTION
**Pull request by wuxingyi**
Signed-off-by: wuxingyi wuxingyi@letv.com

### CODE REVIEW

File: src/include/rados/librados.h [hunk 1]
```diff
@@ -1078,6 +1078,29 @@ CEPH_RADOS_API void rados_object_list_slice(rados_ioctx_t io,
 CEPH_RADOS_API int rados_objects_list_open(rados_ioctx_t io,
                                            rados_list_ctx_t *ctx);
 
+
+ /**
+ * Start listing objects in a specific pg
+ *
+ * @param io the pool to list from
+ * @param ctx the handle to store list context in
```
Comment (tchaikov) [inline src/include/rados/librados.h]: could you fix the copy & paste errors here?

File: src/osdc/Objecter.h [hunk 2]
```diff
@@ -1406,6 +1406,7 @@ class Objecter : public md_config_obs_t, public Dispatcher {
     // is acquired before sending the first OP and released upon receiving
     // the last op reply.
     int ctx_budget;
+    bool listsinglepg;
```
Comment (tchaikov) [inline src/osdc/Objecter.h]: `listing_single_pg`.

File: src/include/rados/librados.h (hunk 1 above)
Comment (wuxingyi) [inline src/include/rados/librados.h]: sure, fixed~

File: src/osdc/Objecter.h (hunk 2 above)
Comment (wuxingyi) [inline src/osdc/Objecter.h]: fixed~

File: src/tools/rados/rados.cc [hunk 3]
```diff
@@ -2009,6 +2017,97 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
...
+    if (!pool_name) {
+      cerr << "pool name was not specified" << std::endl;
+      ret = -1;
+      goto out;
+    }
+
+    if (!pgid)
+    {
+               cerr << "pgid was not specified" << std::endl;
+               ret = -1;
```
Comment (theanalyst) [inline src/tools/rados/rados.cc]: whitespace here seems to be extra

Comment (wuxingyi) [inline src/tools/rados/rados.cc]: sure, fixed~

**Comment by wuxingyi** [comment]
@tchaikov , any further comments?

**Comment by dachary** [comment]
jenkins test this please (jenkins hung)

**Comment by wuxingyi** [comment]
hi, @dachary , I rebased the code and the test seems to work now

**Comment by tchaikov** [comment]
@jdurgin @liewegas  do we want this new API in librados?

**Comment by liewegas** [comment]
I think the concept here is fine, but the recent pg listing API changes should make the implementation much simpler.  All we really need is a function that generates two iterators for the start and end of a particular PG.  Those can be used with either the nobjects or newer enumerate APIs (the cursors are convertible).

**Pull request closed without merging**